### PR TITLE
More memory

### PR DESCRIPTION
**Pull request by bcleveland1022**
Added the ability for user to increase the maximum memory the application is allowed to use.  Also defaulted to half the systems memory when user hasn't set a configuration.

### CODE REVIEW

**Review by schmidtk — CHANGES_REQUESTED** [review]

File: app/src/main.js [hunk 1]
```diff
@@ -81,6 +82,12 @@ const createMainWindow = () => {
   });
 };
 
+const maxMemory = getMaximumMemory();
+log.info('Setting applications maximum memory to ' + maxMemory + ' MB.');
+app.commandLine.appendSwitch('js-flags', '--max-old-space-size=' + maxMemory);
+
+global.sharedObj = {prop1: null};
```
[the remark this answers is not in the document]
Comment (bcleveland1022) [reply app/src/main.js]: Yes I removed it.

File: app/src/memconfig.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+const fs = require('fs');
+const electron = require('electron');
+const log = require('electron-log');
+
+const memoryConfigFileName = '/maxmemory.txt';
```
[the remark this answers is not in the document]
Comment (bcleveland1022) [reply app/src/memconfig.js]: changed it to .maxmemory

File: app/src/memconfig.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Gets the configured maximum memory for the system.
+ * @return {Number} The maximum memory for the system in MB.
+ */
+const getMaximumMemory = () => {
+  // Divide by half system memory and convert to MB.
+  let appMemory = process.getSystemMemoryInfo().total / 2048 | 0;
```
[the remark this answers is not in the document]
Comment (bcleveland1022) [reply app/src/memconfig.js]: Made them consts.

File: app/src/memconfig.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  const userDataPath = (electron.app || electron.remote.app).getPath('userData');
+  const memoryConfigPath = userDataPath + memoryConfigFileName;
```
[the remark this answers is not in the document]
Comment (bcleveland1022) [reply app/src/memconfig.js]: Using path.join.

**Review by schmidtk — APPROVED** [review]

File: app/src/relauncher.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+const electron = require('electron');
+const log = require('electron-log');
+
+const appEnv = require('./appenv.js');
+
+/**
+ * The electron app.
+ */
+const app = (electron.app || electron.remote.app);
```
Comment (schmidtk) [inline app/src/relauncher.js:9]: Given relaunch doesn't work properly in dev and `isDev` won't be correct outside the main process, I don't think we should try using `electron.remote.app` here.

**Review by schmidtk — CHANGES_REQUESTED** [review]

File: app/src/relauncher.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+const electron = require('electron');
```
Comment (schmidtk) [inline app/src/relauncher.js:1]:
```suggestion
const {app} = require('electron');
```

File: app/src/relauncher.js (hunk 5 above)
Comment (bcleveland1022) [reply app/src/relauncher.js]: Just did the below change.

File: app/src/relauncher.js (hunk 6 above)
Comment (bcleveland1022) [reply app/src/relauncher.js]: done.

**Review by jsalankey — APPROVED** [review]

**Pull request merged**